sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala

```diff
@@ -831,12 +831,11 @@ abstract class ToTimestamp
         }
       case StringType =>
         val zid = ctx.addReferenceObj("zoneId", zoneId, classOf[ZoneId].getName)
-        val locale = ctx.addReferenceObj("locale", Locale.US)
         val tf = TimestampFormatter.getClass.getName.stripSuffix("$")
         nullSafeCodeGen(ctx, ev, (string, format) => {
           s"""
             try {
-              ${ev.value} = $tf$$.MODULE$$.apply($format.toString(), $zid, $locale)
+              ${ev.value} = $tf$$.MODULE$$.withStrongLegacy($format.toString(), $zid)
```

Comment:
quick question, why `SIMPLE_DATE_FORMAT` alone has `withStrongLegacy` - is it because `SIMPLE_DATE_FORMAT` is an enum, and tricky for the codegen part to reference? We could just simply define string variables for them too. Seems like it's better to keep it consistent to one side as this legacy code part is already pretty difficult to follow and read.

Reply:
same question. we already have the `LENIENT_SIMPLE_DATE_FORMAT` enum.

Reply:
and method name `withStrongLegacy` looks confusing. We haven't decided to go legacy by here.

Reply:
>  is it because SIMPLE_DATE_FORMAT is an enum, and tricky for the codegen part to reference?

Yes, let me figure out how to use Scala enum from Java

Reply:
I removed `withStrongLegacy` by enum